operator/src/main/java/org/keycloak/operator/Utils.java

```diff
@@ -77,4 +80,35 @@ public static <T extends HasMetadata> Optional<T> getByName(Class<T> clazz, Func
         return ies.get(new ResourceID(nameFunction.apply(primary), primary.getMetadata().getNamespace()));
     }
 
+    /**
+     * Set resources requests/limits for Keycloak container
+     * </p>
+     * If not specified in the Keycloak CR, set default values from operator config
+     */
+    public static void addResources(ResourceRequirements resource, Config config, Container kcContainer) {
+        final ResourceRequirements resourcesSpec = Optional.ofNullable(resource).orElseGet(ResourceRequirements::new);
+
+        // sets the min boundary when the spec is not present
+        final var requests = Optional.ofNullable(resourcesSpec.getRequests()).orElseGet(HashMap::new);
+
+        final var requestsMemory = requests.get("memory");
+        final var defaultRequestsMemory = config.keycloak().resources().requests().memory();
+
+        // Validate 'requests' memory
+        if (requestsMemory != null) {
+            var specifiedMemoryIsLessThanDefault = requestsMemory.getNumericalAmount().intValue() < defaultRequestsMemory.getNumericalAmount().intValue();
+            if (specifiedMemoryIsLessThanDefault) {
+                Log.debugf("Provided 'requests' memory ('%s') is less than used default value ('%s'). Use it in your risk, as Keycloak performance might be degraded.", requestsMemory, defaultRequestsMemory);
```

Comment:
@shawkins Is it ok, or do you suggest to use the `KeycloakStatusAggregator` ?

Reply:
Yes, it's just awkward to use currently because it's associated with the top level reconciliation of the keycloak cr, not any of the dependents.